### PR TITLE
JDK-8301224: Replace NULL with nullptr in share/gc/shared/

### PR DESCRIPTION
**Pull request by jdksjolen**
Hi, this PR changes all occurrences of NULL to nullptr for the subdirectory share/gc/shared/. Unfortunately the script that does the change isn't perfect, and so we
need to comb through these manually to make sure nothing has gone wrong. I also review these changes but things slip past my eyes sometimes.

Here are some typical things to look out for:

1. No changes but copyright header changed (probably because I reverted some changes but forgot the copyright).
2. Macros having their NULL changed to nullptr, these are added to the script when I find them. They should be NULL.
3. nullptr in comments and logs. We try to use lower case "null" in these cases as it reads better. An exception is made when code expressions are in a comment.

An example of this:

```c++
// This function returns null
void* ret_null();
// This function returns true if *x == nullptr
bool is_nullptr(void** x);
```

Note how `nullptr` participates in a code expression here, we really are talking about the specific value `nullptr`.

Thanks!

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8301224](https://bugs.openjdk.org/browse/JDK-8301224): Replace NULL with nullptr in share/gc/shared/

### Reviewers
 * [Stefan Karlsson](https://openjdk.org/census#stefank) (@stefank - **Reviewer**) ⚠️ Review applies to [5369babe](https://git.openjdk.org/jdk/pull/12249/files/5369babef911e4328b8791ae898647559a44960f)
 * [Kim Barrett](https://openjdk.org/census#kbarrett) (@kimbarrett - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk pull/12249/head:pull/12249` \
`$ git checkout pull/12249`

Update a local copy of the PR: \
`$ git checkout pull/12249` \
`$ git pull https://git.openjdk.org/jdk pull/12249/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 12249`

View PR using the GUI difftool: \
`$ git pr show -t 12249`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/12249.diff">https://git.openjdk.org/jdk/pull/12249.diff</a>

</details>

### CODE REVIEW

File: src/hotspot/share/gc/shared/c2/barrierSetC2.cpp [hunk 1]
```diff
@@ -363,7 +363,7 @@ void C2Access::fixup_decorators() {
     const TypePtr* adr_type = _addr.type();
     Node* adr = _addr.node();
     if (!needs_cpu_membar() && adr_type->isa_instptr()) {
-      assert(adr_type->meet(TypePtr::NULL_PTR) != adr_type->remove_speculative(), "should be not null");
+      assert(adr_type->meet(TypePtr::nullptr_PTR) != adr_type->remove_speculative(), "should be not null");
```
Comment (jdksjolen) [inline src/hotspot/share/gc/shared/c2/barrierSetC2.cpp:366]: NULL_PTR

**Review by jdksjolen — COMMENTED** [review]
Found only one issue.

**Review by stefank — APPROVED** [review]

**Comment by jdksjolen** [comment]
Passes tier1.

File: src/hotspot/share/gc/shared/c1/barrierSetC1.cpp [hunk 2]
```diff
@@ -224,7 +224,7 @@ void BarrierSetC1::generate_referent_check(LIRAccess& access, LabelObj* cont) {
   // We need to generate code similar to the following...
   //
   // if (offset == java_lang_ref_Reference::referent_offset()) {
-  //   if (src != NULL) {
+  //   if (src != null) {
```
Comment (kimbarrett) [inline src/hotspot/share/gc/shared/c1/barrierSetC1.cpp:227]: This "null" in a comment really should be "nullptr".

File: src/hotspot/share/gc/shared/generationCounters.cpp [hunk 3]
```diff
@@ -67,7 +67,7 @@ GenerationCounters::GenerationCounters(const char* name,
                                        size_t min_capacity, size_t max_capacity,
                                        VirtualSpace* v)
   : _virtual_space(v) {
-  assert(v != NULL, "don't call this constructor if v == NULL");
+  assert(v != nullptr, "don't call this constructor if v == null");
```
Comment (kimbarrett) [inline src/hotspot/share/gc/shared/generationCounters.cpp:70]: nullptr here

File: src/hotspot/share/gc/shared/oopStorage.hpp [hunk 4]
```diff
@@ -99,12 +99,12 @@ class OopStorage : public CHeapObjBase {
   };
 
   // Locks _allocation_mutex.
-  // precondition: ptr != NULL.
+  // precondition: ptr != null.
```
Comment (kimbarrett) [inline src/hotspot/share/gc/shared/oopStorage.hpp:102]: nullptr here

File: src/hotspot/share/gc/shared/oopStorage.hpp [hunk 5]
```diff
@@ -99,12 +99,12 @@ class OopStorage : public CHeapObjBase {
...
   EntryStatus allocation_status(const oop* ptr) const;
 
-  // Allocates and returns a new entry.  Returns NULL if memory allocation
+  // Allocates and returns a new entry.  Returns null if memory allocation
   // failed.  Locks _allocation_mutex.
-  // postcondition: result == NULL or *result == NULL.
+  // postcondition: result == null or *result == null.
```
Comment (kimbarrett) [inline src/hotspot/share/gc/shared/oopStorage.hpp:107]: nullptr here

File: src/hotspot/share/gc/shared/oopStorage.hpp [hunk 6]
```diff
@@ -120,18 +120,18 @@ class OopStorage : public CHeapObjBase {
   // precondition: size > 0.
   // postcondition: result <= min(size, bulk_allocate_limit).
   // postcondition: ptrs[i] is an allocated entry for i in [0, result).
-  // postcondition: *ptrs[i] == NULL for i in [0, result).
+  // postcondition: *ptrs[i] == null for i in [0, result).
```
Comment (kimbarrett) [inline src/hotspot/share/gc/shared/oopStorage.hpp:123]: nullptr here

File: src/hotspot/share/gc/shared/oopStorage.hpp [hunk 7]
```diff
@@ -120,18 +120,18 @@ class OopStorage : public CHeapObjBase {
...
   size_t allocate(oop** ptrs, size_t size);
 
   // Deallocates ptr.  No locking.
   // precondition: ptr is a valid allocated entry.
-  // precondition: *ptr == NULL.
+  // precondition: *ptr == null.
```
Comment (kimbarrett) [inline src/hotspot/share/gc/shared/oopStorage.hpp:128]: nullptr here

File: src/hotspot/share/gc/shared/oopStorage.hpp [hunk 8]
```diff
@@ -120,18 +120,18 @@ class OopStorage : public CHeapObjBase {
...
   void release(const oop* ptr);
 
   // Releases all the ptrs.  Possibly faster than individual calls to
   // release(oop*).  Best if ptrs is sorted by address.  No locking.
   // precondition: All elements of ptrs are valid allocated entries.
-  // precondition: *ptrs[i] == NULL, for i in [0,size).
+  // precondition: *ptrs[i] == null, for i in [0,size).
```
Comment (kimbarrett) [inline src/hotspot/share/gc/shared/oopStorage.hpp:134]: nullptr here

File: src/hotspot/share/gc/shared/oopStorage.hpp [hunk 9]
```diff
@@ -155,9 +155,9 @@ class OopStorage : public CHeapObjBase {
   // - is_alive->do_object_b(*p) must be a valid expression whose value is
   // convertible to bool.
   //
-  // For weak_oops_do, if *p == NULL then neither is_alive nor closure will be
+  // For weak_oops_do, if *p == null then neither is_alive nor closure will be
```
Comment (kimbarrett) [inline src/hotspot/share/gc/shared/oopStorage.hpp:158]: nullptr here

File: src/hotspot/share/gc/shared/oopStorageParState.hpp [hunk 10]
```diff
@@ -124,7 +124,7 @@
 //   - is_alive->do_object_b(*p) must be a valid expression whose value
 //   is convertible to bool.
 //
-//   If *p == NULL then neither is_alive nor cl will be invoked for p.
+//   If *p == null then neither is_alive nor cl will be invoked for p.
```
Comment (kimbarrett) [inline src/hotspot/share/gc/shared/oopStorageParState.hpp:127]: nullptr here, or "*p is null".

**Review by kimbarrett — CHANGES_REQUESTED** [review]
There are a number of changes to something like `*p == null` that I think should either be `*p == nullptr` or `*p is null`.

**Comment by jdksjolen** [comment]
Hi Kim, thanks, I've pushed your suggested changes.

File: src/hotspot/share/gc/shared/oopStorage.hpp [hunk 11]
```diff
@@ -99,12 +99,12 @@ class OopStorage : public CHeapObjBase {
...
 
   // Locks _allocation_mutex.
-  // precondition: ptr != null.
+  // precondition: ptr != nullptr.
   EntryStatus allocation_status(const oop* ptr) const;
 
   // Allocates and returns a new entry.  Returns null if memory allocation
   // failed.  Locks _allocation_mutex.
-  // postcondition: result == null or *result == null.
+  // postcondition: result == null or *result == nullptr.
```
Comment (kimbarrett) [inline src/hotspot/share/gc/shared/oopStorage.hpp:107]: Inconsistent mix of "null" and "nullptr".

**Review by kimbarrett — CHANGES_REQUESTED** [review]

File: src/hotspot/share/gc/shared/oopStorage.hpp (hunk 11 above)
Comment (jdksjolen) [reply src/hotspot/share/gc/shared/oopStorage.hpp]: Thanks!

**Comment by jdksjolen** [comment]
This is still passing tier1, just waiting for approval.

**Review by kimbarrett — APPROVED** [review]
Looks good.

**Comment by jdksjolen** [comment]
Passes tier1.

/integrate

**Pull request closed without merging**